src/python/ensembl/io/genomio/extract_from_gb.py

```diff
@@ -26,33 +45,38 @@
 from Bio.SeqFeature import SeqFeature
 from BCBio import GFF
 
-from collections import Counter
 
+class GBParseError(Exception):
+    """Error when parsing the Genbank file."""
 
-class InputSchema(argschema.ArgSchema):
-    prefix = argschema.fields.String(metadata={"required": True, "description": "prefix to add required"})
-    prod_name = argschema.fields.String(
-        metadata={"required": True, "description": "production name for the species"}
-    )
-    gb_file = argschema.fields.InputFile(
-        metadata={"required": True, "description": "Sequence accession file"}
-    )
 
+class UnsupportedData(Exception):
+    """When an expected data is not supported by the current parser."""
 
-def extract_gb(prefix, prod_name, gb_file):
-    formatter = FormattedFilesGenerator()
-    formatter.set_prefix(prefix)
-    formatter.set_production_name(prod_name)
-    formatter.parse_genbank(gb_file)
 
-    # Output the gff3 file
-    output = {
-        "gff3": formatter.genes_gff,
-        "fasta_dna": formatter.fasta_dna,
-        "fasta_pep": formatter.fasta_pep,
-        "seq_region": formatter.seq_region,
-        "genome_data": formatter.genome,
-    }
+class GenomeFiles:
```

Comment:
After checking the implementation and its usage, it may be worth considering as a future improvement making `class GenomeFiles(dict):`, so your `GenomeFiles` objects are dictionaries on their own (no need for the `to_dict` function).